Formula/w/watchman.rb

```diff
@@ -70,5 +70,6 @@ def install
 
   test do
     assert_equal(version.to_s, shell_output("#{bin}/watchman -v").chomp)
+    system bin/"watchman", "--help"
```

Comment:
@chenrui333 
looks like `--help` actually exits nonzero. If you want something that would repro this issue can try

`watchman version` (instead of `watchman -v`). appears `watchman version` would have repro'd the seg fault.

<img width="457" alt="image" src="https://github.com/user-attachments/assets/6a5c7c26-b4dc-438c-84de-92c35b822613" />

Reply:
yeah, that is good

Reply:
I guess, it can hardly be tested in CI

```
  ==> /opt/homebrew/Cellar/watchman/2024.12.02.00_2/bin/watchman version
  2024-12-27T14:20:37,140: [cli] Failed to open /Users/brew/Library/LaunchAgents/com.github.facebook.watchman.plist for write: Operation not permitted
  Fatal error detected at:
  0   libfolly.0.58.0-dev.dylib           0x0000000104e821d4 _ZN5folly10symbolizer17getStackTraceSafeEPmm + 12
  1   libfolly.0.58.0-dev.dylib           0x0000000104e833b0 _ZN5folly10symbolizer21SafeStackTracePrinter15printStackTraceEb + 80
  2   watchman                            0x0000000102e5ca20 _ZL15log_stack_tracev + 64
  3   watchman                            0x0000000102e5c95c _ZN8watchman3Log13doLogToStdErrEv + 552
  4   watchman                            0x0000000102e5a9e0 _ZN8watchman9Publisher7enqueueEO8json_ref + 692
  5   watchman                            0x0000000102de0324 _ZN8watchman3Log3logIJRA16_KcRA4096_cRA13_S2_NSt3__112basic_stringIcNS9_11char_traitsIcEENS9_9allocatorIcEEEERA2_S2_EEEvNS_8LogLevelEDpOT_ + 384
  6   watchman                            0x0000000102dda52c main + 2184
  7   dyld                                0x000000019c534274 start + 2840
  (safe mode, symbolizer not available)
  Error: watchman: failed
  An exception occurred within a child process:
    BuildError: Failed executing: /opt/homebrew/Cellar/watchman/2024.12.02.00_2/bin/watchman version
```

Reply:
we can followup on it, and just focus on shipping watchman update for now.